src/control/jobs/control_jobs.c

```diff
@@ -663,6 +664,52 @@ static int32_t dt_control_remove_images_job_run(dt_job_t *job)
 }
 
 
+static void delete_file_from_disk(const char *filename)
+{
+  const char *filename_display = NULL;
+  GError *gerror = NULL;
+  GFileInfo *gfileinfo = NULL;
+  gboolean delete_success = FALSE;
+
+  GFile *gfile = g_file_new_for_path(filename);
+  int send_to_trash = dt_conf_get_bool("send_to_trash");
+
+  if (send_to_trash)
+  {
+    delete_success = g_file_trash(gfile, NULL /*cancellable*/, &gerror);
+  }
+  else
+  {
+    delete_success = g_file_delete(gfile, NULL /*cancellable*/, &gerror);
+  }
+
+  if (!delete_success)
```

Comment:
From mail:
* when trashing fails, you silently revert to deleting. That's a bad idea. Please do a modal dialog with a message asking the user if he wants to delete or cancel instead